htdocs/accountancy/class/accountancyexport.class.php

```diff
@@ -399,6 +438,69 @@ public function export(&$TData, $formatexportset)
 				}
 				break;
 		}
+
+		// create and download export file or archive
+		if ($withAttachment == 1) {
+			$error = 0;
+
+			// close export file
+			if ($exportFile) {
+				fclose($exportFile);
+			}
+
+			if (!empty($archiveFullName) && !empty($archivePath) && !empty($archiveFileList)) {
+				// archive files
+				$downloadFileMimeType = 'application/zip';
+				$downloadFileFullName = $archiveFullName;
+				$downloadFilePath = $archivePath;
+
+				// create archive
+				$archive = new ZipArchive();
```

Comment:
Why don't you use dol_compress_dir to build the archive ?

Reply:
@eldy 
I will see how to use it

Reply:
dol_compress_dir() function needs two mandatory parameters :
- inputdir
- outputfile
For "outputfile", I can use "archivePath" variable but I don't have a source directory for "inputdir" parameter.
All documents provided for the compressed file are in different directories (object directory).